src/main/java/io/netty/incubator/codec/http3/Http3ServerPushStreamManager.java

```diff
@@ -0,0 +1,289 @@
+     * @return the {@link Future} that will be notified once the push-stream was opened.
+     */
+    public Future<QuicStreamChannel> newPushStream(long pushId, ChannelHandler handler) {
+        final Promise<QuicStreamChannel> promise = channel.eventLoop().newPromise();
+        newPushStream(pushId, handler, promise);
+        return promise;
+    }
+
+    /**
+     * Returns a new HTTP/3 push-stream that will use the given {@link ChannelHandler}
+     * to dispatch {@link Http3PushStreamFrame}s too. The needed HTTP/3 codecs are automatically added to the
+     * pipeline as well.
+     *
+     * @param pushId for the push stream. This MUST be obtained using {@link #reserveNextPushId()}.
+     * @param handler the {@link ChannelHandler} to add. Can be {@code null}.
+     * @param promise to indicate creation of the push stream.
+     */
+    public void newPushStream(long pushId, ChannelHandler handler, Promise<QuicStreamChannel> promise) {
+        validatePushId(pushId);
+        channel.createStream(QuicStreamType.UNIDIRECTIONAL, pushStreamInitializer(pushId, handler), promise);
+        setupCancelPushIfStreamCreationFails(pushId, promise, channel);
+    }
+
+    /**
+     * Returns a new HTTP/3 push-stream that will use the given {@link ChannelHandler}
+     * to dispatch {@link Http3PushStreamFrame}s too. The needed HTTP/3 codecs are automatically added to the
+     * pipeline as well.
+     *
+     * @param pushId for the push stream. This MUST be obtained using {@link #reserveNextPushId()}.
+     * @param handler the {@link ChannelHandler} to add. Can be {@code null}.
+     * @param bootstrapConfigurator {@link UnaryOperator} to configure the {@link QuicStreamChannelBootstrap} used.
+     * @param promise to indicate creation of the push stream.
+     */
+    public void newPushStream(long pushId, ChannelHandler handler,
+                              UnaryOperator<QuicStreamChannelBootstrap> bootstrapConfigurator,
+                              Promise<QuicStreamChannel> promise) {
+        validatePushId(pushId);
+        QuicStreamChannelBootstrap bootstrap = bootstrapConfigurator.apply(channel.newStreamBootstrap());
+        bootstrap.type(QuicStreamType.UNIDIRECTIONAL)
+                .handler(pushStreamInitializer(pushId, handler))
+                .create(promise);
+        setupCancelPushIfStreamCreationFails(pushId, promise, channel);
+    }
+
+    /**
+     * A {@link ChannelInboundHandler} to be added to the {@link QuicChannel} associated with this
+     * {@link Http3ServerPushStreamManager} to listen to control stream frames.
+     *
+     * @return {@link ChannelInboundHandler} to be added to the {@link QuicChannel} associated with this
+     * {@link Http3ServerPushStreamManager} to listen to control stream frames.
+     */
+    public ChannelInboundHandler controlStreamListener() {
+        return controlStreamListener;
+    }
+
+    private boolean isPushAllowed0(long maxPushId) {
```

Comment:
No need to append `0` in the method name because there's no signature conflict without `0`.